plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2059,6 +2060,13 @@ So if getMinSpeed() returns null we fall back to getSpeed().
         final InputDef input = new InputDef("tablet", "usb");
         devices.addDevice(input);
 
+        // If we're using virtio scsi, then we need to add a virtual scsi controller
+        if (getGuestDiskModel(vmTO.getPlatformEmulator()) == DiskDef.DiskBus.SCSI) {
+            final SCSIDef sd = new SCSIDef((short)0, 0, 0, 9, 0);
+            devices.addDevice(sd);
+            s_logger.debug("Adding scsi definition:\n" + sd.toString());
```

Comment:
Might want to include the name of the VM here as well? And SCSI is written in upper-case.